lib/ts/unit-tests/test_ink_inet.cc

```diff
@@ -144,3 +144,92 @@ TEST_CASE("ats_ip_pton", "[libts][inet][ink_inet]")
   addr.load("c000::ffff:ffff:ffff:ffff:ffff:ffff");
   REQUIRE(addr == upper);
 }
+
+TEST_CASE("inet formatting", "[libts][ink_inet][bwformat]")
+{
+  IpEndpoint ep;
+  ts::string_view addr_1{"[ffee::24c3:3349:3cee:143]:8080"};
+  ts::string_view addr_2{"172.17.99.231:23995"};
+  ts::string_view addr_3{"[1337:ded:BEEF::]:53874"};
+  ts::string_view addr_4{"[1337::ded:BEEF]:53874"};
+  ts::string_view addr_5{"[1337:0:0:ded:BEEF:0:0:956]:53874"};
+  ts::string_view addr_6{"[1337:0:0:ded:BEEF:0:0:0]:53874"};
+  ts::string_view addr_7{"172.19.3.105:49951"};
+  ts::string_view addr_null{"[::]:53874"};
+  ts::LocalBufferWriter<1024> w;
+
+  REQUIRE(0 == ats_ip_pton(addr_1, &ep.sa));
+  w.print("{}", ep);
+  REQUIRE(w.view() == addr_1);
+  w.reset().print("{::p}", ep);
+  REQUIRE(w.view() == "8080");
+  w.reset().print("{::a}", ep);
+  REQUIRE(w.view() == addr_1.substr(1, 24)); // check the brackets are dropped.
+  w.reset().print("[{::a}]", ep);
+  REQUIRE(w.view() == addr_1.substr(0, 26)); // check the brackets are dropped.
+  w.reset().print("[{0::a}]:{0::p}", ep);
+  REQUIRE(w.view() == addr_1); // check the brackets are dropped.
+  w.reset().print("{::^a}", ep);
+  REQUIRE(w.view() == "ffee:0000:0000:0000:24c3:3349:3cee:0143");
+  w.reset().print("{:: ^a}", ep);
```

Comment:
I am going to -1 this PR and other BufferWriter PRs until there is more discussion if this is they way we want to go with formatting and handling strings.

There have been people in the community speaking up on this.  Can you please send an email to the dev@ mailing list.

Reply:
@bryancall  So was @SolidWallOfCode mistaken in thinking there was a consensus in support of python-like formatted output for BufferWriter?